stellargraph/data/unsupervised_sampler.py

```diff
@@ -55,24 +69,25 @@ def __init__(self, G, nodes=None, length=2, number_of_walks=1, seed=None):
         else:
             self.graph = G
 
-        # Instantiate the walker class used to generate random walks in the graph
-        self.walker = UniformRandomWalk(G, seed=seed)
-
-        # This code will enable alternative walker classes
-        # TODO: Enable this code, but figure out how to pass required options to run
-        # if walker is not None:
-        #     if not isinstance(
-        #         walker, UniformRandomWalk
-        #     ):  # only work with Uniform Random Walker at the moment
-        #         raise TypeError(
-        #             "({}) Only Uniform Random Walks are possible".format(
-        #                 type(self).__name__
-        #             )
-        #         )
-        #     else:
-        #         self.walker = walker(G, seed=seed)
-        # else:
-        #         self.walker = UniformRandomWalk(G, seed=seed)
+        if walker is not None:
+            # only work with UniformRandomWalker and BiasedRandomWalker at the moment
+            if not isinstance(walker, UniformRandomWalk) and not isinstance(
+                walker, BiasedRandomWalk
+            ):
+                raise TypeError(
+                    "({}) Only the UniformRandomWalks and BiasedRandomWalks are possible".format(
+                        type(self).__name__
+                    )
+                )
+            else:
+                self.walker = walker
+        else:
+            self.walker = UniformRandomWalk(G, seed=seed)
+
+        if isinstance(self.walker, BiasedRandomWalk):
```

Comment:
I might look into refactoring `BiasedRandomWalk` (and other walker classes) to optionally take these parameters upon construction  rather than in `run`, so that we can avoid doing this - I think this make more sense than having to pass in the same set of parameters for each call to `run` and it seems generally useful to be able to instantiate and pass around a walker object that already has its hyper-parameters filled in?

Reply:
Hi Kevin,

In my earlier version,  I ever tried to pass the hyperparameter (p, q) of `BiasedRandomWalk` into its initiate function rather than pass them into the `run` function. I will make changes to be consistent with the new interface.

Reply:
@daokunzhang This should be landing soon: https://github.com/stellargraph/stellargraph/pull/1187

Reply:
And this is related: https://github.com/stellargraph/stellargraph/pull/1179

Reply:
OK, I will merge it.